### PR TITLE
Add scheduled removal date to obsolete API methods

### PR DESCRIPTION
**Pull request by skamphuis**
#2359 

## Summary
Used Regex search & replace to add notice "Scheduled removal in v11.0.0."
 to all obsolete messages.

### CODE REVIEW

File: DNN Platform/Library/Common/Utilities/XmlUtils.cs [hunk 1]
```diff
@@ -824,7 +824,7 @@ public static string XPathLiteral(string value)
             sb.Append(")");
             return sb.ToString();
         }
-        [Obsolete("This method is obsolete.")]
+        [Obsolete("This method is obsolete.. Scheduled removal in v11.0.0.")]
```
Comment (valadas) [inline DNN Platform/Library/Common/Utilities/XmlUtils.cs]:
```suggestion
        [Obsolete("This method is obsolete. Use .Net XmlDocument.Load instead. Scheduled removal in v11.0.0.")]
```

File: DNN Platform/Library/Data/DataProvider.cs [hunk 2]
```diff
@@ -3948,7 +3948,7 @@ public virtual int GetContentWorkflowStateUsageCount(int stateId)
             return ExecuteScalar<int>("GetContentWorkflowStateUsageCount", stateId);
         }
 
-        [Obsolete("Deprecated in Platform 7.4.0")]
+        [Obsolete("Deprecated in Platform 7.4.0. Scheduled removal in v11.0.0.")]
```
Comment (valadas) [inline DNN Platform/Library/Data/DataProvider.cs]: I am not familiar enough with this, but we should provide some basic guidance on what to replace it with.

File: DNN Platform/Library/Entities/Host/Host.cs [hunk 3]
```diff
@@ -1223,7 +1223,7 @@ public static bool ShowCriticalErrors
         ///   Defaults to 1
         /// </remarks>
         /// -----------------------------------------------------------------------------
-        [Obsolete("Deprecated in 8.0.0")]
+        [Obsolete("Deprecated in 8.0.0. Scheduled removal in v11.0.0.")]
```
Comment (valadas) [inline DNN Platform/Library/Entities/Host/Host.cs]: Not sure if the site log is removed or just methods replace, but some basic explanation should be given to developers on what to do. Same for the others in this namespace

File: DNN Platform/Library/Obsolete/PortalController.cs [hunk 4]
```diff
@@ -87,7 +87,7 @@ public partial class PortalController
         }
 
         [EditorBrowsable(EditorBrowsableState.Never)]
-        [Obsolete("Deprecated in DotNetNuke 7.3.0.")]
+        [Obsolete("Deprecated in DotNetNuke 7.3.0.. Scheduled removal in v11.0.0.")]
```
Comment (valadas) [inline DNN Platform/Library/Obsolete/PortalController.cs]: Need some details on what replaces this method

File: DNN Platform/Tests/DNN.Integration.Test.Framework/Controllers/ModuleController.cs [hunk 5]
```diff
@@ -20,7 +20,7 @@ public static class ModuleController
         private const string PermissionKeyMarker = @"$[permission_key]";
         private const string FriendlyNameMarker = @"$[friendly_name]";
 
-        [Obsolete("typo in name, this should be removed")]
+        [Obsolete("typo in name, this should be removed. Scheduled removal in v11.0.0.")]
```
Comment (valadas) [inline DNN Platform/Tests/DNN.Integration.Test.Framework/Controllers/ModuleController.cs]:
```suggestion
        [Obsolete("typo in name, use AddModulePermission instead. Scheduled removal in v11.0.0.")]
```

File: DNN Platform/Library/Entities/Portals/PortalInfo.cs [hunk 6]
```diff
@@ -754,7 +754,7 @@ public string RegisteredRoleName
         #endregion
 
         [XmlIgnore]
-        [Obsolete("Deprecated in DNN 6.0.")]
+        [Obsolete("Deprecated in DNN 6.0.. Scheduled removal in v11.0.0.")]
```
[the remark this answers is not in the document]
Comment (sleupold) [reply DNN Platform/Library/Entities/Portals/PortalInfo.cs]: Get Role from PortalInfo.RegisteredRoleID

[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Library/Entities/Portals/PortalInfo.cs]: Thanks

[the remark this answers is not in the document]
Comment (valadas) [reply DNN Platform/Library/Entities/Portals/PortalInfo.cs]:
```suggestion
        [Obsolete("Deprecated in DNN 6.0. Use POrtalInfo.RegisteredroleID instead. Scheduled removal in v11.0.0.")]
```

File: DNN Platform/Library/Common/Utilities/XmlUtils.cs (hunk 1 above)
Comment (ohine) [reply DNN Platform/Library/Common/Utilities/XmlUtils.cs]: I just pushed this change.

File: DNN Platform/Library/Entities/Portals/PortalInfo.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (ohine) [reply DNN Platform/Library/Entities/Portals/PortalInfo.cs]: Line # 757 isn't `RegisteredRoleName`, it's TimeZoneOffet so this change doesn't make sense.

[the remark this answers is not in the document]
Comment (ohine) [reply DNN Platform/Library/Entities/Portals/PortalInfo.cs]: I did just remove the extra period however.

[the remark this answers is not in the document]
Comment (skamphuis) [reply DNN Platform/Library/Entities/Portals/PortalInfo.cs]: >             I did just remove the extra period however.

Hmm, I might have introduced the extra period a few more times. Searching for ".. Scheduled removal in v11.0.0." might be worth a try.

**Pull request merged**